### PR TITLE
ADD: Groupby plotting capability.

### PR DESCRIPTION
**Pull request by rcjackson**
Addresses #402. I have now added a wrapper around the Display object that allows the user to group their display by time periods (right now it's (year, month, day, hour, minute, second)). 

The groupby can be initiated with any display object using groupby = display.group_by('year') if you want to group the plot by year. Then you would use groupby.plot_group('plot_func_name', datastream, keyword arguments for the plotting function in the Display object)

For example, this does wind rose data plots grouped by day for the data rose plot example:
obj = act.io.armfiles.read_netcdf(act.tests.EXAMPLE_MET_WILDCARD)

# Create Plot Display
display = act.plotting.WindRoseDisplay(obj, figsize=(15, 15), subplot_shape=(3, 3))
groupby = display.group_by('day')
groupby.plot_group('plot_data', None, dir_field='wdir_vec_mean', spd_field='wspd_vec_mean',
                   data_field='temp_mean', num_dirs=12, plot_type='line')
image

### CODE REVIEW

**Review by mgrover1 — APPROVED** [review]
Great addition! I like the class used here, and the tests/example you put together.

**Comment by zssherman** [comment]
@rcjackson The test failure is unrelated and fixed in Ken's PR. But yeah this looks great!

**Comment by zssherman** [comment]
@rcjackson Is there anyway to fix the example plot, the overlap of title and windrose top marker?

**Comment by mgrover1** [comment]
a `plt.tight_layout()` should fix it?

**Comment by zssherman** [comment]
Rerunning tests as kens PR should fix this now.

File: examples/plotting/plot_days.py [hunk 1]
```diff
@@ -0,0 +1,26 @@
+"""
+Calculate and plot wind rose plots separated by day.
+-----------------------------------------------------
+
+Example of how to read in MET data and plot histograms
+of wind speed and temperature grouped by day.
+
+Author: Adam Theisen
```
Comment (AdamTheisen) [inline examples/plotting/plot_days.py:8]: @rcjackson I think you need to update the author on this one.

**Comment by zssherman** [comment]
Resetting tests to match changes in Ken's PR

**Comment by AdamTheisen** [comment]
Looks like a cropscape test is failing so they must have updated the DB. We can change that test to be = Soybeans.

**Comment by rcjackson** [comment]
Fixed the plotting example to move the axes ticks inside the rose.

**Comment by AdamTheisen** [comment]
@rcjackson I'm running into some issues when I try and test this out on plotting daily temperatures.  Code is 

```
obj = act.io.armfiles.read_netcdf(act.tests.sample_files.EXAMPLE_MET_WILDCARD)
print(obj['time'].values)
display = act.plotting.TimeSeriesDisplay(obj, subplot_shape=(3, 3))
groupby = display.group_by('day')
groupby.plot_group('plot', None, field='temp_mean')
plt.show()
```

But the only plot that has data is the first one.  Am I missing something?
image

**Comment by AdamTheisen** [comment]
Note, I got the same when trying to do a month groupby on all of 2022 data.

image

**Comment by rcjackson** [comment]
I think the default behavior of TimeSeriesDisplay to set all x axes limits the same was causing it. In this case, we don't want them to be the same or they would all just be for the first time period. I put some code to override that behavior in the GroupBy when it detects a TimeSeriesDisplay to set to the min and max time for each group.
image

**Review by zssherman — COMMENTED** [review]
@rcjackson Do we know if this will occur elsewhere in other plotting modules?

File: act/plotting/plot.py [hunk 2]
```diff
@@ -278,3 +279,126 @@ def add_colorbar(self, mappable, title=None, subplot_index=(0,), pad=None,
...
+        """
+        self.display = display
+        self._groupby = {}
+        num_groups = 0
+        datastreams = list(display._obj.keys())
+        for key in datastreams:
+            self._groupby[key] = display._obj[key].groupby('time.%s' % units)
+            num_groups = max([num_groups, len(self._groupby[key])])
+
+    def plot_group(self, func_name, dsname, **kwargs):
```
Comment (mgrover1) [inline act/plotting/plot.py:320]: Is there anyway we can automatically fill the dsname to be None?

**Comment by rcjackson** [comment]
> @rcjackson Do we know if this will occur elsewhere in other plotting modules?
None of the other modules use time as an x-axis, so it should not affect them.

File: act/plotting/plot.py (hunk 2 above)
Comment (rcjackson) [reply act/plotting/plot.py]: Done in new commit!

**Review by zssherman — APPROVED** [review]

**Comment by zssherman** [comment]
Looks good, if no one else has anything to add, will merge this.

**Comment by AdamTheisen** [comment]
Just tested and looks good!  Thanks for the updates @rcjackson !  @zssherman I think we're good to go

**Pull request merged**